image-assets/upload-image-to-aws.sh

```diff
@@ -0,0 +1,233 @@
+# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-ami-boot-mode.html
+# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/launch-instance-boot-mode.html
+
+set -e
+set -o pipefail
+
+checkArguments() {
+  if [ $# -lt 1 ]; then
+    echo "Error: You need to specify the cloud image to upload."
+    echo "Usage: $0 <cloud-image>"
+    exit 1
+  fi
+
+  local file="$1"
+
+  if [ ! -f "$file" ]; then
+    echo "Error: File '$file' does not exist."
+    exit 1
+  fi
+
+  if ! file "$file" | grep -q 'DOS/MBR boot sector'; then
+    echo "Error: File '$file' is not a raw image."
+    exit 1
+  fi
+}
+
+checkEnvVars() {
+  if [ -z "$AWS_PROFILE" ] || [ -z "$AWS_REGION" ] || [ -z "$AWS_S3_BUCKET" ]; then
+    echo "Error: AWS_PROFILE, AWS_REGION and AWS_S3_BUCKET environment variables must be set."
+    exit 1
+  fi
+}
+
+AWS() {
+  aws --profile $AWS_PROFILE --region $AWS_REGION "$@"
+}
+
+# https://docs.aws.amazon.com/vm-import/latest/userguide/required-permissions.html#vmimport-role
+ensureVmImportRole() {
+  (AWS iam list-roles | jq -r '.Roles[] | select(.RoleName | contains("vmimport")) | .RoleName' | grep -q "vmimport" && echo "vmimport role found. All good.") || {
+    echo "Creating vmimport role"
+
+    export AWS_PAGER="" # Avoid being dropped to a pager
+    AWS iam create-role --role-name vmimport --assume-role-policy-document file://<(cat <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "vmie.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {
+        "StringEquals": {
+          "sts:ExternalId": "vmimport"
+        }
+      }
+    }
+  ]
+}
+EOF
+)
+
+  #  AWS iam attach-role-policy --role-name vmimport --policy-arn arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM
+
+    AWS iam put-role-policy --role-name vmimport --policy-name vmimport --policy-document file://<(cat <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetBucketLocation",
+        "s3:GetBucketAcl",
+        "s3:GetObject",
+        "s3:PutObject",
+        "s3:ListBucket"
+      ],
+      "Resource": [
+        "arn:aws:s3:::$AWS_S3_BUCKET",
+        "arn:aws:s3:::$AWS_S3_BUCKET/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CancelConversionTask",
+        "ec2:CancelExportTask",
+        "ec2:CreateImage",
+        "ec2:CreateInstanceExportTask",
+        "ec2:CreateTags",
+        "ec2:ExportImage",
+        "ec2:ImportInstance",
+        "ec2:ImportVolume",
+        "ec2:StartInstances",
+        "ec2:StopInstances",
+        "ec2:TerminateInstances",
+        "ec2:ImportImage",
+        "ec2:ImportSnapshot",
+        "ec2:CancelImportTask",
+        "ec2:ModifySnapshotAttribute",
+        "ec2:CopySnapshot",
+        "ec2:RegisterImage",
+        "ec2:Describe*"
```

Comment:
Only these are mentioned in the docs (see the link at the top of the function).

I will try to remove the rest and see if things still work.

Reply:
removed and everything works.